### PR TITLE
Change data model restaurant to location

### LINKED ISSUE #8: rename and review restaurant data model to location

### PR DESCRIPTION
**Pull request by kc-leung**
fix #8 

- Created Data Model called location and locationImage

I have decided to make a new schema for locationImage for a better performance

### CODE REVIEW

**Review by coreyjjames — CHANGES_REQUESTED** [review]
Looks good just comparing with the UML and we had a few more fields then you have here.

1)  "OptionalInformation" field, which I am thinking would be a place to hold optional notes about the location. We can change the name to something like "locationNotes" if you want.

2) "categoyId" field, which will store the id for the type Category.

3)  "locationMark" field, which will store the id for the type Mark. (Note: UML says its a Set of marks but that I think is wrong a location would only save one mark.

**Comment by kc-leung** [comment]
> 
> 
> Looks good just comparing with the UML and we had a few more fields then you have here.
> 
>     1. "OptionalInformation" field, which I am thinking would be a place to hold optional notes about the location. We can change the name to something like "locationNotes" if you want.
> 
>     2. "categoyId" field, which will store the id for the type Category.
> 
>     3. "locationMark" field, which will store the id for the type Mark. (Note: UML says its a Set of marks but that I think is wrong a location would only save one mark.

Will add OptionalInformation as AdditionalInformation and locationMarkSet. Also, I was thinking to bring up in the coming meeting changing category relation to mark instead of location. You can just think about the reason of why and why not of that suggestion and we can discuss on thursday

File: models/location.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+// create Location Schema & model
+const LocationSchema = new Schema({
+  locationId: {
+    type: Schema.Types.ObjectId,
+    required: [true, "locationId is required to save Location"]
+  },
+  groupId: {
```
Comment (coreyjjames) [inline models/location.js:10]: UML relates location to Mark, not Group. Then the group will end up relating to Mark.

**Review by coreyjjames — CHANGES_REQUESTED** [review]
Ya we can leave out categories for now and talk about it in the meeting

**Comment by kc-leung** [comment]
> 
> 
> Ya we can leave out categories for now and talk about it in the meeting

I am leaving the groupID since it is there previously I need to test it first with the routes and services so I can know the reason of having it or not

**Review by coreyjjames — APPROVED** [review]

**Pull request merged**